malduck/extractor/config_builder.py

```diff
@@ -0,0 +1,538 @@
+        """
+        Capabilities of the malware that are disabled.
+        """
+        return self.push_config(dict(capability_disabled=capability_disabled))
+
+    def add_campaign_id(self, *campaign_id: str) -> None:
+        """
+        Server/campaign id for malware.
+        """
+        return self.push_config(dict(campaign_id=campaign_id))
+
+    def add_identifier(self, *identifier: str) -> None:
+        """
+        UUID/Identifiers for deployed instance of malware.
+        """
+        return self.push_config(dict(identifier=identifier))
+
+    def add_decoded_strings(self, *string: str) -> None:
+        """
+        Decoded string from within malware.
+
+        Usually there is a lot of strings so contents of this field
+        might be put in separate object during further processing.
+        """
+        return self.push_config(dict(decoded_strings=string))
+
+    def add_password(self, *password: str) -> None:
+        """
+        Any password extracted from the binary
+        """
+        return self.push_config(dict(password=password))
+
+    def add_mutex(self, *mutex: str) -> None:
+        """
+        Mutex to prevent multiple instances
+        """
+        return self.push_config(dict(mutex=mutex))
+
+    def add_pipe(self, *pipe: str) -> None:
+        """
+        Pipe name used for communication
+        """
+        return self.push_config(dict(pipe=pipe))
+
+    def set_sleep_delay(self, sleep_delay: int) -> None:
+        """
+        Time to sleep/delay execution
+        """
+        return self.push_config(dict(sleep_delay=sleep_delay))
+
+    def add_inject_exe(self, *inject_exe: str) -> None:
+        """
+        Name of executable to inject into
+        """
+        return self.push_config(dict(inject_exe=inject_exe))
+
+    def add_binary(
+        self,
+        filename: str,
+        data: bytes,
+        datatype: Optional[ExtractorModel.Binary.TypeEnum] = None,
+        encryption: Optional[ExtractorModel.Binary.Encryption] = None,
+        other: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Extracted binary data e.g. additional files.
+
+        :param filename: Name that identifies a file
+        :param data: Contents of the binary
+        :param datatype: Type of the binary
+        :param encryption: Encryption used to protect the binary
+        :param other: Other metadata
+
+        .. note::
+
+            ``filename`` is not a field of Maco model but it's usually necessary to
+            identify binary, so we decided to make it required. Value of this field
+            will be put in "other".
+        """
+        binary = ExtractorModel.Binary(
+            data=data,
+            datatype=datatype,
+            encryption=encryption,
+            other={**(other or {}), "filename": filename},
+        )
+        return self.push_config(dict(binaries=[binary]))
+
+    def add_ftp(
+        self,
+        username: Optional[str] = None,
```

Comment:
is everything really optional? (here and below) at least hostname sound important i guess

Reply:
I blindly folllowed Maco scheme there, but yeah, if we don't have alternative like ip vs domain, we should treat hostname as required.